GitCommands/Git/GitRevision.cs

```diff
@@ -120,9 +120,17 @@ private void OnPropertyChanged(string propertyName)
             PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
         }
 
+        /// <summary>
+        /// Returns a value indicating whether <paramref name="id"/> is a valid SHA-1 hash.
+        /// </summary>
+        /// <remarks>
+        /// To be valid the string must contain exactly 40 lower-case hexadecimal characters.
+        /// </remarks>
+        /// <param name="id">The string to validate.</param>
+        /// <returns><c>true</c> if <paramref name="id"/> is a valid SHA-1 hash, otherwise <c>false</c>.</returns>
         public static bool IsFullSha1Hash(string id)
         {
-            return Regex.IsMatch(id, Sha1HashPattern);
+            return Sha1HashRegex.IsMatch(id);
```

Comment:
In case this isn't easy to spot, the old code used the Regex pattern `[a-f\d]{40}` and the new code uses `^[a-f\d]{40}$`.

Reply:
A-F should have been accepted too, bux if it was not before, there is no difference.

Reply:
I noticed in #4556 that git always produces lower case SHA-1 hashes. I guess the only case we'd want to accept mixed case is when parsing user-entered data.